client/src/pages/rollup/Show.js

```diff
@@ -82,6 +117,164 @@ const GQL_EMAIL_ROLLUP = gql`
+
+  const advisorOrgGraphData = useMemo(() => {
+    return Object.values(graphData?.advisorOrgReports || {}).sort((a, b) =>
+      a.org.shortName > b.org.shortName ? 1 : -1
+    )
+  }, [graphData])
+  const principalOrgGraphData = useMemo(() => {
+    return Object.values(graphData?.principalOrgReports || {}).sort((a, b) =>
+      a.org.shortName > b.org.shortName ? 1 : -1
+    )
+  }, [graphData])
+
+  const displayedGraphData =
+    orgType === Organization.TYPE.ADVISOR_ORG
+      ? advisorOrgGraphData
+      : principalOrgGraphData
+
+  if (done) {
+    return result
+  }
+
+  const CHART_ID = "reports_by_organization"
+  const barColors = {
+    cancelled: "#ec971f",
+    published: "#75eb75"
+  }
+  const legendCss = {
+    width: "14px",
+    height: "14px",
+    display: "inline-block"
+  }
+  return (
+    <div className="scrollable-y">
+      <ContainerDimensions>
+        {({ width }) => (
+          <>
+            <ButtonToggleGroup value={orgType} onChange={setOrgType}>
+              <Button
+                value={Organization.TYPE.ADVISOR_ORG}
+                variant="outline-secondary"
+              >
+                {pluralize(Settings.fields.advisor.org.name)}
+              </Button>
+              <Button
+                value={Organization.TYPE.PRINCIPAL_ORG}
+                variant="outline-secondary"
+              >
+                {pluralize(Settings.fields.principal.org.name)}
+              </Button>
+            </ButtonToggleGroup>
+            <DailyRollupChart
+              width={width}
+              chartId={CHART_ID}
+              data={displayedGraphData}
+              onBarClick={setOrg}
+              tooltip={d => `
+              <h4>${d.org.shortName}</h4>
+              <p>Published: ${d.published}</p>
+              <p>Cancelled: ${d.cancelled}</p>
+              <p>Click to view details</p>
+            `}
+              barColors={barColors}
+            />
+          </>
+        )}
+      </ContainerDimensions>
+
+      <div className="graph-legend">
+        <div style={{ ...legendCss, background: barColors.published }} />{" "}
+        Published reports:&nbsp;
+        <strong>
+          {displayedGraphData.reduce((acc, org) => acc + org.published, 0)}
+        </strong>
+      </div>
+      <div className="graph-legend">
+        <div style={{ ...legendCss, background: barColors.cancelled }} />{" "}
+        Cancelled engagements:&nbsp;
+        <strong>
+          {displayedGraphData.reduce((acc, org) => acc + org.cancelled, 0)}
+        </strong>
+      </div>
+    </div>
+  )
+}
+
+Chart.propTypes = {
+  queryParams: PropTypes.object,
+  pageDispatchers: PageDispatchersPropType,
+  setOrg: PropTypes.func
+}
+
+const generateChartDataFromAllReports = (allReports, orgFilterUuid) => {
+  return allReports.reduce(
+    (acc, r) => {
+      const topLevelAdvisorOrg = r.advisorOrg.ascendantOrgs[0]
+      const topLevelPrincipalOrg = r.principalOrg.ascendantOrgs[0]
```

Comment:
`r.advisorOrg` and `r.principalOrg` could be null here (e.g. for Draft reports).